### ISSUE TITLE
WP_REST_Request:get_params()/get_param() returns route...

### ISSUE THREAD

**Issue by Shelob9** [WP-API/WP-API#1621]
Not sure if this was intended, but it's annoying me. In my callback, where `$request` is `WP_REST_Request` object and I do this:

```
$params = $request->get_params();
$zero = $request->get_param( 0 );
```

I get a 0 index in `$params` and `$zero` has the route URL. I don't see why it <em>should</em> be a param, it's not registered on the endpoint, and its redundant to what I can get with the `get_url_params()` method.

**Comment by joehoyle** [comment]
Hmm I'm not sure if this was ever intentional or not, perhaps @rmccue knows?

**Comment by Shelob9** [comment]
@joehoyle I don't know about intentional. I would argue it's not useful information in that context. That information is available via other means that make more sense.

**Comment by danielbachhuber** [comment]
Closing in favor of https://core.trac.wordpress.org/ticket/34647, as this needs to be addressed before WP 4.4 ships.
